MuscleModel/ReadFromModel/getMuscleInfo.m

```diff
@@ -106,22 +91,48 @@
 [B,A] = butter(Misc.f_order_lMT,Misc.f_cutoff_lMT/(fs/2));
 DatStore.LMT = filtfilt(B,A,LMT_raw);
 
-% store various informaiton in the DatStore structure
+% store infomration in the DatStore structure
 DatStore.MuscleNames = Misc.MuscleNames;
 DatStore.DOFNames    = Misc.DofNames;
 DatStore.nMuscles    = length(Misc.MuscleNames);
 DatStore.nDOF        = length(Misc.DofNames);
-DatStore.q_exp       = IK_data.data(IK_inds,DOF_inds+1);        % +1 for time vector
 
+%% Filter IK
+
+% load joint kinematics
+[~,Misc.trialName,~]=fileparts(IK_path);
+IK_data=importdata(IK_path);
+if ~isfield(IK_data,'colheaders')
+    IK_data.colheaders=strsplit(IK_data.textdata{end});
+end
+
+% select the IK information between the selected time frames
+t_IK=IK_data.data(:,1);     t_IK=round(t_IK*10000)/10000; 
+ind0=find(t_IK>=Misc.time(1),1,'first'); ind_end=find(t_IK<=Misc.time(2),1,'last');
+IK_inds=ind0:ind_end;
+
+% filter the kinematics and kinetics
+fs=1/mean(diff(t_IK));
+[B, A] = butter(Misc.f_order_IK, Misc.f_cutoff_IK/(fs/2));
+[~,DOFS] = size(IK_data.data);
+for i = 2:DOFS         % filtering time vector not needed (start from 2)
+    IK_data.data(:,i) = filtfilt(B,A,IK_data.data(:,i));
+end
+
+%% Filter ID
 % Get the ID data
 ID_data=importdata(ID_path);
 t_ID=ID_data.data(:,1); t_ID=round(t_ID*10000)/10000;
 
 % get the ID index
-ID_header=ID_data.colheaders;
-ID_Header_inds=size(DOF_inds);
+if ~isfield(ID_data,'colheaders')
+    ID_data.colheaders=strsplit(ID_data.textdata{end});
+end
+ID_header=ID_data.colheaders;     IK_header = IK_data.colheaders;
+ID_Header_inds=zeros(size(DOF_inds));  IK_Header_inds = zeros(size(DOF_inds));
```

Comment:
@MaartenAfschrift, Tuur's issue was related to the use of size. Looks good to me now that you added `zeros(...)`. I think he had a dimension issue when selecting only one dof.